src/_base/harness/config/pipeline.yml

```diff
@@ -87,7 +87,7 @@ command('app deploy <environment>'):
     #!bash(harness:/helm)|=
     set -o pipefail
     cd "${ENVIRONMENT}"
-    doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml
+    (umask 0077 && doctl -t "$DO_ACCESS_TOKEN" kubernetes cluster kubeconfig show "$CLUSTER" > kubectl.config.yaml)
```

Comment:
For the [record](https://github.com/inviqa/harness-base-php/pull/710#discussion_r876336860): `umask` removes the bits that are allowed on a file on creation, so e.g. 0007 means if a file created with 666 permissions then it gets 660 permissions. You can see it easily with (umask 0077 && touch testfile) that it gets 600 permissions.